src/QLExpression.actor.cpp

```diff
@@ -55,13 +55,13 @@ ACTOR Future<Optional<std::pair<Standalone<StringRef>, DataValue>>> getArrayAnce
 	return Optional<std::pair<Standalone<StringRef>, DataValue>>();
 }
 
-static Future<Void> doPathExpansion(PromiseStream<Reference<IReadContext>> const& promises,
-                                    Standalone<StringRef> const& queryPath,
-                                    Reference<IReadContext> const& document,
-                                    int const& arrayLookStart,
-                                    bool const& isOuter,
-                                    bool const& expandLastArray,
-                                    bool const& imputeNulls);
+ACTOR static Future<Void> doPathExpansion(PromiseStream<Reference<IReadContext>> promises,
```

Comment:
Why not simply moving the actual implementation of `doPathExpansion` to here to avoid forward declaration totally?

Reply:
`doPathExpansion` is already in this file. But, `doArrayExpansion` and `doPathExpansion` depend on one another, so unfortunately  forward declaration is unavoidable until we rewrite this logic.

Reply:
Oh yeah I missed that dependency...